### PR TITLE
Removed query-string, use goto svelte for url state changes

### PR DESCRIPTION
**Pull request by Ak4zh**
- Added new environment variable POST_CATEGORIES
- Added post filters using post categories
- Removed the use of `query-string` lib as the same task can be achieved with svelte '$page.url.searchParams`
- Removed all urlState functions and changed to use svelte `goto` instead.

### CODE REVIEW

File: src/routes/blog/+page.svelte
Comment (sw-yx) [inline src/routes/blog/+page.svelte:75]: this code is required for URLs to be copy pastable

eg when i search something on my site it shows up in the URL, and can be copy pasted

image


but on this PR it doesnt

image

File: src/lib/siteConfig.js
Comment (sw-yx) [inline src/lib/siteConfig.js:11]:
```suggestion
export const POST_CATEGORIES = ['Blog'] // Other categories you can consider adding: Talks, Tutorials, Snippets, Podcasts, Notes...
```

**Comment by sw-yx** [comment]
appreciate the PR! left some thoughts before merging, curious what you think. let me know if you are interested in being a maintainer as well - i just ask that you send every change in as a PR so that we can discuss first and I can keep track

File: src/routes/blog/+page.svelte
Comment (Ak4zh) [reply src/routes/blog/+page.svelte:75]: [14691a7](https://github.com/sw-yx/swyxkit/pull/97/commits/14691a773585345df798423901576f72ecf016a8)
Fixed the URLs to be updated with search filter.

**Comment by Ak4zh** [comment]
I think URL parameter name should be changed as below for a more clear naming:
- `show` to `categories` or `tags` (I prefer tags as it's shorter)
- `filter` to `search` or `keyword` (I prefer keyword here as search gives a feeling of path rather that a param)

Let me know if that looks ok.

**Comment by sw-yx** [comment]
almost! when you go to here: https://deploy-preview-97--swyxkit.netlify.app/blog?filter=test&show=Blog

the "show" code is dropped somehow

image

**Comment by sw-yx** [comment]
i tried fixing it but couldnt find the source of the bug - i can get back to it in future, thank you for the contribution, it is very valued.

**Comment by Ak4zh** [comment]
> i tried fixing it but couldnt find the source of the bug - i can get back to it in future, thank you for the contribution, it is very valued.

Fixed.
It seems it was happening because both getting the param get value and set were reactive and running at same time.

File: README.md
Comment (sw-yx) [inline README.md:115]:
```suggestion
export const POST_CATEGORIES = ['Blog'] // Other categories you can consider adding: Talks, Tutorials, Snippets, Podcasts, Notes...
```

File: README.md
Comment (sw-yx) [inline README.md:52]: added attribution to you @Ak4zh  :)

**Review by sw-yx — APPROVED** [review]

**Comment by sw-yx** [comment]
thank you so much!

**Pull request merged**
